components/camel-jpa/src/main/java/org/apache/camel/component/jpa/JpaProducer.java

```diff
@@ -202,10 +203,18 @@ protected void processQuery(Exchange exchange, EntityManager entityManager) {
 
     @SuppressWarnings("unchecked")
     private void configureParameters(Query query, Exchange exchange) {
-        int maxResults = getEndpoint().getMaximumResults();
+        final int maxResults = Objects.requireNonNullElse(
+                exchange.getIn().getHeader(JpaConstants.JPA_MAXIMUM_RESULTS, Integer.class),
```

Comment:
You can use the getHeader that takes a default value to void the Objects method. This is the standard we use in Camel.